Lokaskil_postgresql.md

```diff
@@ -0,0 +1,707 @@
+,House Blackmyre
+The Westerlands,House Lannett
+The Westerlands,House Lanny
+Iron Islands,House Netley
+The Riverlands,House Nutt
+,House Peat
+The Riverlands,House Perryn
+The Reach,House Pommingham
+The Vale,House Ruthermont
+The Riverlands,House Fisher
+The North,House Branfield
+The Stormlands,House Wensington
+The North,House Knott
+The Riverlands,House Harroway of Harrenhal
+The North,House Flint of Breakstone Hill
+Iron Islands,House Goodbrother of Downdelving
+Dorne,House Shell
+The North,House Glenmore
+The Riverlands,House Teague
+The Vale,House Brightstone
+Dorne,House Dryland
+Dorne,House Briar
+Dorne,House Brook
+Dorne,House Brownhill
+,House Langward
+Dorne,House Holt
+Dorne,House Lake
+The Vale,House Shell
+The Reach,House Appleton of Appleton
+The North,House Lake
+The Vale,House Lipps
+,House Pyne
+,House Cave
+Iron Islands,House Goodbrother of Orkmont
+The Stormlands,House Rogers of Amberly
+The North,House Holt
+Dorne,House Hull
+The North,House Frost
+The Vale,House Hersy of Newkeep
+The North,House Marsh
+The Reach,House Stackhouse
+,House Dargood
+Dorne,House Wade
+The Westerlands,House Algood
+The North,House Stane of Driftwood Hall
+The Crownsland,
+Gift,
+
+## Útskýring á lið 2
+
+Nokkur atriði úr kóðanum til að skoða.
+
+`WITH gagntaek_vorpun AS (
+    SELECT
+        l.gid AS location_id,
+        l.name AS location_name,
+        h.id AS house_id,
+        h.name AS house_name,
+        l.summary AS summary
+    FROM
+        atlas.locations l
+    JOIN
+        got.houses h ON l.name = ANY(h.seats)
+    WHERE
+        h.region = 'The North'
+)`
+
+-Hér býr kóðinn til kortlagningu (gagntaek_vorpun) staðsetningar á hús út frá staðsetningarheiti
+ sem gæti passað við hvaða sæti hússins sem er (ANY(h.sæti)).
+
+`X AS (
+    SELECT *, regexp_match(summary, house_name) FROM gagntaek_vorpun
+),
+y AS (
+    SELECT
+        CASE WHEN house_name LIKE concat('%',location_name) THEN 1 ELSE 0 END as better_match,
+        *
+    FROM x
+),
+z AS (
+    SELECT row_number() OVER (PARTITION BY location_id ORDER BY better_match DESC) as rank, * FROM y
+)`
+
+Hér er verið að að vinna með þegar það koma tvívegis mötch og unnið úr hvernig eigi að velja betra
+matchið. Þá er þetta gert til að grípa tvítekningar.
+
+`g AS (
+    SELECT house_id, location_id FROM z WHERE rank = 1
+)`
+
+- skipun sem velur besta matchið. Eða 1 frekar  en 0.
+
+`INSERT INTO baratheon.tables_mapping (house_id, location_id)
+SELECT house_id, location_id FROM g
+ON CONFLICT (house_id) DO UPDATE SET location_id = excluded.location_id;`
+
+-Varpar inn í baratheon.tables_mapping töfluna og leysir árekstra með því að uppfæra
+ núverandi færslur ef house_id er þegar til í töflunni.
+
+`SELECT
+    h.name AS house_name,
+    h.id AS house_id,
+    l.name AS location_name,
+    l.gid AS location_id
+FROM
+    gagntaek_vorpun otm
+JOIN
+    got.houses h ON otm.house_id = h.id
+JOIN
+    atlas.locations l ON otm.location_id = l.id
+WHERE
+    h.region = 'The North';`
```

Comment:
Þetta er ekki það sama og er í SQL skjalinu ykkar, því þetta er þar, og þið eruð með `id` dálk í locations en hann er `gid`

````suggestion
```sql
SELECT 
    h.name AS house_name, 
    h.id AS house_id, 
    l.name AS location_name, 
    l.gid ASlocation_id 
FROM 
    "baratheon.tables_mapping" otm 
JOIN 
    got.houses h ON otm.houses_id = h.id 
JOIN
    atlas.locations l ON otm.location_id = l.gid 
WHERE 
    h.region = 'The North';
```
````